### ISSUE TITLE
QueryParam and Err in OpenApi are not flexible enough

### ISSUE THREAD

**Issue by immno** [salvo-rs/salvo#593]
1. QueryParam only supports basic types, not struct.
2. I can only use StatusError and ParseError. When an error is found in the parameter, I cannot customize the return format.
3. As mentioned above, I want to use `validator` crate , but the `response` is not what I want, so the validator is not built into the custom `Extractible`.

When I wanted to extend a `QueryParam` that supports `struct`, I found that `EndpointArgRegister` was not easy to implement using the `schema` of the `struct` and could not be split out.
I saw the code related to handling OpenAPI and Err, which limits my custom response processing.
``` rust                   
extract_ts.push(quote! {
        let #id: #ty = match <#ty as #salvo::Extractible>::extract_with_arg(__macro_generated_req, #idv).await {
            Ok(data) => {
                data
            },
            Err(e) => {
                #salvo::__private::tracing::error!(error = ?e, "failed to extract data in endpoint macro");
                res.render(#salvo::http::errors::StatusError::bad_request().brief(
                    "Failed to extract data in endpoint macro."
                ).cause(e));
                return;
            }
        };
    });
```
Thanks

**Comment by immno** [comment]
Do you have any relevant plans? I want to let the team use salvo in this next project. By the way, are there any relevant communication groups, such as WeChat?

**Comment by chrislearn** [comment]
> Do you have any relevant plans? I want to let the team use salvo in this next project. By the way, are there any relevant communication groups, such as WeChat?

my wechat: chrislearn

**Comment by chrislearn** [comment]
Fixed by #606
